frontends/ol-util/src/factories.ts

```diff
@@ -7,7 +7,7 @@ type Factory<T, U = never> = (overrides?: Partial<T>, options?: U) => T
 const makePaginatedFactory =
   <T>(makeResult: Factory<T>) =>
     (
-      count: number,
+      { count, pageSize }: { count: number; pageSize?: number },
```

Comment:
`makePaginatedFactory` takes a factory function and returns a factory that creates paginated results, `{ count, previous, next, results }`. 

Previously `count` and `results.length` were always the same, which isn't flexible enough.